c_tests/tests/compile_calls.c

```diff
@@ -0,0 +1,31 @@
+// Compiler:
+// Run-time:
+
+// Check that basic trace compilation works.
+// FIXME An optimising compiler can remove all of the code between start/stop
+// tracing.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk_testing.h>
+
+void f() { int b = 1; }
+
+int main(int argc, char **argv) {
+  int res = 0;
```

Comment:
This variable is pointless as it's not assigned to. Either we should remove it, or have `f` return a value which we assign to `res`?

Reply:
it's passed by reference to `__yktrace_start_tracing`.

Reply:
(which mutates it)

Reply:
So it does... but now I'm not sure why!?

Reply:
I think this is the way lukas is getting stuff into and out of the trace (for now).

But lukas, if this does mutate `res` (to 0) we should initialise it to something other than 0? Otherwise how do we know it has been mutated?

Reply:
Ah, in this test `res` could go. It's not needed as this test only checks that an inlined function doesn't crash the trace compiler. In the next commit however, we alter this test to write the return value of `f()` to it, so we might as well leave it here too.

Reply:
OK (not ideal, but probably not worth fixing). Can we also say why `res` is passed to `__yktrace_start_tracing`?

Reply:
Ah, yes, that definitely deserves a clarifying comment which probably should have gone into #353.

Reply:
OK, can we add a comment commit into this PR because I must admit that I missed it there!